xbmc/music/MusicDatabase.cpp

```diff
@@ -2180,6 +2223,101 @@ bool CMusicDatabase::ClearArtistLastScrapedTime(int idArtist)
   return ExecuteQuery(strSQL);
 }
 
+bool CMusicDatabase::AddArtistVideoLinks(const CArtist& artist)
```

Comment:
Assuming that there will be more songs for any given album artist than there will be videolink it would be better to loop through the links rather than the songs when doing the match up.

Querying the songs for a videolink (when matching on album artist and song title there could be more than one song) with something like 
```
SELECT idSong FROM song 
WHERE strMusicBrainzTrackID = '%s' OR 
(EXISTS (SELECT 1 FROM album_artist WHERE idAlbum = song.idAlbum AND idArtist = %i ) 
AND strTitle LIKE '%%s%%')

```
Then updating song and art tables accordingly.

Reply:
Ah well, matching on one or the other is because some songs can be remixes or re-releases and have the same title but different mbids  E.G. ZZ Top - Tush has an mbid of both `fe8de756-c0b0-468f-9304-e92821f0d2fb` and `699fe8a8-7c7e-44dc-a592-edf307888165`. Both are in my music library but only one is returned from TADB for the music video.  Similarly, their track 'Legs' has at least three different mbids, `b7072e8d-39f8-4e7c-8c30-4317ca590d2e`, `19bac3b7-5452-400c-bf43-54469c501c80` and `23c16131-8304-46b3-a380-67348170fac4`.  Technically, the third is an edited version but the other two are the same track.

However, for the purposes of adding a video link to play a song, I would expect all tracks named 'Legs' by 'ZZ Top' to link to a video if one exists.  The fact that I have a slightly shorter edited version is not the point and in any case, if there was an mbid link to the shorter video (if it even exists) then it would match that first.

So yes, there are good reasons why the matching is done the way it is, otherwise we will just get users saying "I have song x by artist y and there is a link on tadb for a video but Kodi doesn't see it".

I have multiple logs that can show the matching at work if you'd like to peruse them.